File: Data Structures & Algorithms/target-sum/submission-0.cpp
class Solution {
public:
int solve(int i, vector<int>& nums, int target, int res) {
        if(i >= nums.size()){
            if(res == target)return 1;
            return 0;
        }
        return solve(i + 1, nums, target, res + nums[i]) + solve(i + 1, nums, target, res - nums[i]);
    }
    int findTargetSumWays(vector<int>& nums, int target) {
        return solve(0, nums, target, 0);
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-0.go
func maxSubArray(nums []int) int {
    res := nums[0]
    currSum := 0
    for _, e := range nums {
        if currSum < 0 {
            currSum = 0
        }
        currSum += e
        if res < currSum {
            res = currSum
        }
    }
    return res
}
